ui/packages/catalog-ui-search/src/main/webapp/react-component/container/theme-settings/theme-settings.tsx

```diff
@@ -0,0 +1,353 @@
+const saveSpacingChanges = (spacingMode: any) => {
+  let preferences = getPreferences()
+  let newSpacingMode = spacingMode
+  preferences.get('theme').set('spacingMode', newSpacingMode)
+  getPreferences().savePreferences()
+}
+
+const saveAnimationChanges = (animationMode: any) => {
+  let preferences = getPreferences()
+  let newAnimationMode = animationMode
+  preferences.set('animation', newAnimationMode)
+  getPreferences().savePreferences()
+}
+
+const saveHoverPreviewChanges = (hoverValue: any) => {
+  let preferences = getPreferences()
+  let newHoverPreview = hoverValue
+  preferences.set('hoverPreview', newHoverPreview)
+  getPreferences().savePreferences()
+}
+
+const saveThemeChanges = (themeValue: any) => {
+  let preferences = getPreferences()
+  let newTheme = themeValue
+  preferences.get('theme').set('theme', newTheme)
+  getPreferences().savePreferences()
+}
+
+const saveCustomColorVariable = (colorVariable: any, selectedValue: any) => {
+  let preferences = getPreferences()
+  let newValue = selectedValue
+  preferences.get('theme').set(colorVariable, newValue)
+  getPreferences().savePreferences()
+}
+
+class ThemeSettings extends React.Component<
+  {},
+  {
+    fontSizeModel: any
+    spacingModeModel: any
+    animationModel: any
+    hoverPreviewModel: any
+    themeModel: any
+    customToggle: boolean
+    customColors: any[]
+  }
+> {
+  constructor(props: any) {
+    super(props)
+    const themeScale = ThemeUtils.getZoomScale(getFontSize())
+    this.state = {
+      fontSizeModel: new Property({
+        isEditing: true,
+        label: 'Zoom Percentage',
+        value: [themeScale],
+        min: 62,
+        max: 200,
+        units: '%',
+        type: 'RANGE',
+      }),
+      spacingModeModel: new Property({
+        isEditing: true,
+        enum: [
+          {
+            label: 'Comfortable',
+            value: 'comfortable',
+          },
+          {
+            label: 'Cozy',
+            value: 'cozy',
+          },
+          {
+            label: 'Compact',
+            value: 'compact',
+          },
+        ],
+        value: [getSpacingMode()],
+        id: 'Spacing',
+      }),
+      animationModel: new Property({
+        isEditing: true,
+        label: 'Animation',
+        value: [getAnimationMode()],
+        enum: [
+          {
+            label: 'On',
+            value: true,
+          },
+          {
+            label: 'Off',
+            value: false,
+          },
+        ],
+        id: 'Animation',
+      }),
+      hoverPreviewModel: new Property({
+        isEditing: true,
+        label: 'Preview Full Image on Hover',
+        value: [user.getHoverPreview()],
+        enum: [
+          {
+            label: 'On',
+            value: true,
+          },
+          {
+            label: 'Off',
+            value: false,
+          },
+        ],
+        id: 'Full Image on Hover',
+      }),
+      themeModel: new Property({
+        isEditing: true,
+        enum: [
```

Comment:
It might be nice to pull these enum lists out above the component since they take up a lot of space and don't provide functional information.